wasm/src/main/scala/ir2wasm/LibraryPatches.scala

```diff
@@ -0,0 +1,111 @@
+package wasm.ir2wasm
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import org.scalajs.ir.ClassKind._
+import org.scalajs.ir.Names._
+import org.scalajs.ir.Position
+import org.scalajs.ir.Trees._
+import org.scalajs.ir.Types._
+
+import org.scalajs.linker.interface.IRFile
+import org.scalajs.linker.interface.unstable.IRFileImpl
+
+import wasm.utils.TestIRBuilder._
+import wasm.utils.MemClassDefIRFile
+
+/** Patches that we apply to the standard library classes to make them wasm-friendly. */
```

Comment:
Is it correct that this patch implementation is temporary? At the moment only a few (js-interoped) methods are referenced, so the number of patches is minimal.
However, as we try to compile more of the standard library for wasm in the future, the number of patches is likely to increase. It seems that manually writing each patch in scalaj's IR could become quite tedious.

In the future, instead of patches, could we prepare implementations specifically for wasm in addition to for JS, such as for `java.lang.String`, and then use those implementations at link time if the target is wasm?

Reply:
It depends for which methods, but eventually they should all be dealt with without that kind of patching, yes:

* For some methods (looking at `Character.toString`), I think we could already upstream the change to Scala.js core (possibility offsetting it by a better intrinsic/codegen).
* For others, we'll probably have to keep them here until we can merge Wasm support in Scala.js core. At that point, we'll be able to:
  * for some methods (for example `numberHashCode`), have an `if (isWasm) ... else ...` logic, where `isWasm` is a link-time constant;
  * for others, deal with them as *intrinsics* in the optimizer (e.g., `Double.doubleToLongBits`).

Reply:
Thanks, that makes sense!